packages/react-router-dom/docs/guides/query-string-parsing.md

```diff
@@ -0,0 +1,61 @@
+# Query String Parsing
+
+In earlier versions of React Router we provided out-of-the-box support for query string parsing. Since there are too many different parsing-algorithms out there and people had problems with an opinionated version, it's been removed from React Router since v4.
```

Comment:
This was not removed because there are too many opinions. `history` just has no reason to include a `query` object in location strings because it has no use for them. All it wants is a `search` string that it can concatenate with a `pathname` and a `hash`.

Also, the problem with query parsing/stringifying isn't that there are too many algorithms. I think that it is more accurate to say that there is no right/wrong way to structure a query string. That leads to a lot of different solutions to support different structures. In turn, that would require any code that supports query parsing to allow the user to provide their own function. That requires `history` to structure itself to do something it doesn't actually care about. 

Honestly, it would be quite simple to add just query string parsing to `history`, but there are two sides to this coin. Removing query object support has more to do with query object serialization. Instead of `history` being able to just concatenate the `search` string with the `pathname` and `hash`, it instead would have to stringify the `query` object and possibly figure out how to merge it with the `search` string. 

That has issues with redundant data and how to merge query objects and search strings and I think that that is the bigger reason why automatic parsing/stringifying was removed.